frame/election-provider-multi-phase/test-staking-e2e/src/mock.rs

```diff
@@ -0,0 +1,745 @@
+	const SNAPSHOT_MAXIMUM_VOTERS: u32 = 1000;
+	const MINER_MAXIMUM_VOTERS: u32 = 1000;
+	const MAXIMUM_TARGETS: u32 = 300;
+}
+
+pub struct OtherSessionHandler;
+impl traits::OneSessionHandler<AccountId> for OtherSessionHandler {
+	type Key = testing::UintAuthorityId;
+
+	fn on_genesis_session<'a, I: 'a>(_: I)
+	where
+		I: Iterator<Item = (&'a AccountId, Self::Key)>,
+		AccountId: 'a,
+	{
+	}
+
+	fn on_new_session<'a, I: 'a>(_: bool, _: I, _: I)
+	where
+		I: Iterator<Item = (&'a AccountId, Self::Key)>,
+		AccountId: 'a,
+	{
+	}
+
+	fn on_disabled(_validator_index: u32) {}
+}
+
+impl sp_runtime::BoundToRuntimeAppPublic for OtherSessionHandler {
+	type Public = testing::UintAuthorityId;
+}
+
+pub struct ExtBuilder {
+	validator_count: u32,
+	minimum_validator_count: u32,
+	invulnerables: Vec<AccountId>,
+	has_stakers: bool,
+	initialize_first_session: bool,
+	pub min_nominator_bond: Balance,
+	min_validator_bond: Balance,
+	balance_factor: Balance,
+	status: BTreeMap<AccountId, StakerStatus<AccountId>>,
+	stakes: BTreeMap<AccountId, Balance>,
+	stakers: Vec<(AccountId, AccountId, Balance, StakerStatus<AccountId>)>,
+}
+
+impl Default for ExtBuilder {
+	fn default() -> Self {
+		Self {
+			validator_count: 2,
+			minimum_validator_count: 0,
+			balance_factor: 1,
+			invulnerables: vec![],
+			has_stakers: true,
+			initialize_first_session: true,
+			min_nominator_bond: ExistentialDeposit::get(),
+			min_validator_bond: ExistentialDeposit::get(),
+			status: Default::default(),
+			stakes: Default::default(),
+			stakers: Default::default(),
+		}
+	}
+}
+
+parameter_types! {}
+
+impl ExtBuilder {
+	pub fn build(self) -> sp_io::TestExternalities {
+		sp_tracing::try_init_simple();
+		let mut storage =
+			frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+		let _ = pallet_balances::GenesisConfig::<Runtime> {
+			balances: vec![
+				(1, 10 * self.balance_factor),
+				(2, 20 * self.balance_factor),
+				(3, 300 * self.balance_factor),
+				(4, 400 * self.balance_factor),
+				// controllers
+				(10, self.balance_factor),
+				(20, self.balance_factor),
+				(30, self.balance_factor),
+				(40, self.balance_factor),
+				(50, self.balance_factor),
+				(60, self.balance_factor),
+				(70, self.balance_factor),
+				(80, self.balance_factor),
+				(90, self.balance_factor),
+				(100, self.balance_factor),
+				(200, self.balance_factor),
+				// stashes
+				(11, self.balance_factor * 1000),
+				(21, self.balance_factor * 2000),
+				(31, self.balance_factor * 3000),
+				(41, self.balance_factor * 4000),
+				(51, self.balance_factor * 5000),
+				(61, self.balance_factor * 6000),
+				(71, self.balance_factor * 7000),
+				(81, self.balance_factor * 8000),
+				(91, self.balance_factor * 9000),
+				(101, self.balance_factor * 10000),
+				(201, self.balance_factor * 20000),
+				// This allows us to have a total_payout different from 0.
+				(999, 1_000_000_000_000),
+			],
```

Comment:
Should we store these in individual vecs outside, such that their role is more clear? 

if you have taken this from the staking pallet, staking has among the worse test setups. I suggest inspiring from EPM.